### PR TITLE
Add `cursor::skip_whitespace_backwards` as a complement to `cursor::s…

### PR DESCRIPTION
**Pull request by agathazeren**
…kip_whitespace`

This replaces the elisp `(skip-chars-backward " \r\t\n")`

### CODE REVIEW

**Review by Sntx626 — APPROVED** [review]
Looks good as well.

At first I thought that maybe the function should state that it can return None, though I guess since it's implemented into the cursor it's a whole different topic of replacing that with something like a result.

Code does what PR promises and PR is justified.
Can be merged.

File: rust/element/src/cursor.rs [hunk 1]
```diff
@@ -171,6 +171,20 @@ impl<'a> Cursor<'a> {
         self.pos()
     }
 
+    /// Skip over space, tabs and newline characters, going backwards
+    /// Cursor position is set before the previous non-whitespace char
+    pub fn skip_whitespace_backwards(&mut self) -> usize {
+        while let Some(c) = self.get_prev_char() {
+            if !(c.is_whitespace()) {
```
Comment (appetrosyan) [inline rust/element/src/cursor.rs:194]:
```suggestion
            if !c.is_whitespace() {
```
No need for extra parens.

Comment (appetrosyan) [reply rust/element/src/cursor.rs:194]: I'd also suggest transposing the if/else clauses. In Rust it's more common to see short circuiting/base case first.

File: rust/element/src/cursor.rs [hunk 2]
```diff
@@ -607,6 +621,25 @@ mod test {
...
+        cursor = Cursor::new(&rope2, rope2.len() - 1);
+        cursor.skip_whitespace_backwards();
+        assert_eq!(cursor.get_next_char().unwrap(), '!');
+
+        // Skipping all the remaining whitespace results in invalid cursor at the beginning of the rope
+        let rope3 = " ";
+        cursor = Cursor::new(&rope3, 1);
+        cursor.skip_whitespace_backwards();
+        assert_eq!(None, cursor.get_prev_char());
+    }
```
Comment (appetrosyan) [inline rust/element/src/cursor.rs:664]: Good unit test.

**Review by appetrosyan — APPROVED** [review]
LGTM

**Pull request merged**